pkg/buildid/buildid.go

```diff
@@ -28,11 +28,13 @@ import (
 	"github.com/parca-dev/parca-agent/pkg/elfreader"
 )
 
-func BuildID(path string) (string, error) {
-	f, err := elf.Open(path)
+func BuildID(f *elf.File, path string) (string, error) {
```

Comment:
```suggestion
func BuildID(f *elf.File) (string, error) {
```

Better to eliminate the `path` to unify. Even if it's passed to a helper function we should refactor it as well.

Reply:
Made the [changes](https://github.com/parca-dev/parca-agent/pull/1001/commits/bdf32954e9e29e137ea4f8e1ef790bf93f340077)- but they seem a bit overkill to me(although cleaner). Let me know if there might be a better approach I am missing out on.

Reply:
https://github.com/parca-dev/parca-agent/pull/1001#discussion_r1029046143